pepper-apis/docs/specification/src/components/schemas.yml

```diff
@@ -378,6 +398,12 @@ DateValue:
       type: integer
       nullable: true
       minimum: 0
+FileAnswerValue:
```

Comment:
Given that this is sent via the `PATCH`, how about we style this after the `FileUploadRequest` and return a `FileUploadResponse` in the reply to the `PATCH`? Saves clients from making another upload request to an alternate endpoint, the server has the bucket id

Reply:
Not quite following your suggestion or understand your point about saving another upload request. Could you clarify?  Just to be super-clear about intention here: in the response to the  "create file upload URL request", we include an `uploadId` (along with the Google Bucket signed URL and resumable flag). The intent of client PATCH ing the upload id is to inform server that client has completed the upload that it authorized earlier and link in the back-end the file location.

Reply:
Gist to the rescue!

https://gist.github.com/bskinner/fc715e80cad6e615a84a805feab85e84

While I was typing this up, I had another thought, and realized I might be approaching it from the wrong direction- what if we used a `201 NO CONTENT` instead? Stick the upload URL in the `Location` header?

Downside, we'd lose the `validationFailures` and `blockVisibility` updates for that pass. Given that the only things needed are a `name` and `fileSize`, I think we can get by without the `validationFailures`. It's more likely the `blockVisibility` _might_ change, though- other blocks could change their state based on if a file was uploaded or not.

I think that could be mitigated by adding the platform requirement that a `File` question is not able to affect another `block`'s visiblity. It feels a bit hacky, but good enough for a version 1.

Reply:
So the idea would be that the upload URL would be obtained by client PATCHing  file name, size, etc and upload URL would be in either the PATCH response body or in Location header?

Reply:
Yep! My thoughts are:
* It reduces number of requests the client needs
* Feels like it works within our current model
* Gives both the server and client a little more context around the file upload (could be useful for caching down the road, for example)
* In Pepper, it might help the implementation of the bucket creation to be a little more isolated

I am leaning more towards the "modify the `PATCH` body" approach, though. It's a less drastic departure from the current API, and I feel it fits in place a bit better. The `201` would need to be re-designed from its base if we ever enable submitting batched answers in a `PATCH`